### PR TITLE
Create the default root API address path

### LINKED ISSUE #8184: Default rootful path to podman.sock not present
Found while creating slides.

We need to make sure that the default path is created `os.MkdirAll(...)`. I had a quick look but couldn't find the right place.
```
$ sudo ./bin/podman system service 
ERRO[0000] /run/podman/podman.sock                      
Error: unable to create socket unix:/run/podman/podman.sock: listen unix /run/podman/podman.sock: bind: no such file or directory
```
@jwhonce PTAL

### PR DESCRIPTION
**Pull request by Luap99**
Fixes #8184

### CODE REVIEW

File: cmd/podman/system/service.go [hunk 1]
```diff
@@ -131,20 +131,17 @@ func resolveAPIURI(_url []string) (string, error) {
...
-		if _, err := os.Stat(filepath.Dir(socketDir)); err != nil {
-			if os.IsNotExist(err) {
-				if err := os.Mkdir(filepath.Dir(socketDir), 0755); err != nil {
-					return "", err
-				}
-			} else {
-				return "", err
-			}
+		socketPath := filepath.Join(xdg, "podman", socketName)
+		if err := os.MkdirAll(filepath.Dir(socketPath), 0755); err != nil {
```
Comment (mheon) [inline cmd/podman/system/service.go:135]: Should this be more restrictive than 755? We may want to 700 for security reasons

**Comment by mheon** [comment]
@vrothberg @jwhonce PTAL

File: cmd/podman/system/service.go [hunk 2]
```diff
@@ -131,20 +131,17 @@ func resolveAPIURI(_url []string) (string, error) {
 		if srvArgs.Varlink {
 			socketName = "io.podman"
 		}
-		socketDir := filepath.Join(xdg, "podman", socketName)
-		if _, err := os.Stat(filepath.Dir(socketDir)); err != nil {
```
Comment (vrothberg) [inline cmd/podman/system/service.go:135]: I don't think that'll fix https://github.com/containers/podman/issues/8184 entirely as it affects root (as well).

Comment (Luap99) [reply cmd/podman/system/service.go:135]: root is handled below in the default case

Comment (vrothberg) [reply cmd/podman/system/service.go:135]: Time to move away from the screen :) Thanks!

File: cmd/podman/system/service.go (hunk 1 above)
Comment (vrothberg) [reply cmd/podman/system/service.go:135]: 700 would make sense. Others shouldn't access it.

**Comment by rhatdan** [comment]
LGTM
/approve

**Comment by TomSweeneyRedHat** [comment]
Nice rework @Luap99 .
One test is failing, I've restarted, the only output was a 404 error from it, which I'm assuming is a flake.
LGTM once tests are hip.

**Pull request merged**